src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidUserDataException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidUserDataException("Password can't be null");
+        }
+        for (User person:Storage.people) {
+            if (person.getLogin().equals(user.getLogin())) {
+                throw new InvalidUserDataException("You cannot add two users "
+                        + "with the same login to the storage");
+            }
+        }
+        if (user.getLogin().length() < MINIMUM_LENGTH) {
+            throw new InvalidUserDataException("Login length is " + user.getLogin().length()
+                    + ", the login cannot contain less than 6 characters");
```

Comment:
The check for the login length should use a different constant than the password length check to avoid confusion and allow for different constraints on login and password lengths in the future.